### PR TITLE
[docs] Add GenRL asset preparation recipe

### PR DESCRIPTION
**Pull request by Abecid**
## Purpose

Adds a reproducible GenRL HPSv3 + VideoAlign training recipe for Wan 2.1 T2V 1.3B on the modular `fastvideo/train` stack.

This PR makes the GenRL run reproducible without relying on a local Modal launcher or git submodules. Runtime assets are prepared through a public helper script, reward dependencies are pinned, and the vendored HPSv3/VideoAlign runtime code is aligned with the reward implementations used by the previous successful GenRL run.

Fixes: N/A

## Changes

- Added `examples/train/prepare_genrl_assets.py` to prepare:
  - GenRL filtered prompt JSONL files
  - `KwaiVGI/VideoReward` checkpoint under `.cache/VideoReward`
  - optional reward-model preflight via `--check-rewards`
- Added `examples/train/requirements-genrl.txt` with the GenRL reward-stack dependency pins.
- Moved the GenRL Wan config to `examples/train/configs/rl/wan/genrl_hpsv3_videoalign.yaml`.
- Updated `examples/train/README.md` with a non-Modal reproduction path.
- Vendored only the HPSv3 and VideoAlign runtime files needed by the GenRL reward wrappers.
- Aligned vendored reward runtimes with the exact successful-run upstream revisions:
  - HPSv3: `a2eb2ef2c7b5d91a566347a5825cf6d872122149`
  - VideoAlign: `aba26b658fec7d9fd30c295187b548ea673c8769`
- Added reward-head load checks so HPSv3/VideoAlign fail fast instead of scoring with randomly initialized reward heads.
- Added prompt dataset validation for missing files, Git LFS pointer files, malformed JSONL, and non-object JSON entries.
- Kept `modal_train_genrl.py` local-only and ignored; it is not part of this PR.

## Reproduction / Training Command

Install the GenRL reward-stack pins after the editable FastVideo install:

    pip install -r examples/train/requirements-genrl.txt

Prepare prompts and reward checkpoints:

    python examples/train/prepare_genrl_assets.py \
      --prompt-dir .cache/genrl_filtered_prompts \
      --genrl-cache-dir .cache/GenRL \
      --videoalign-dir .cache/VideoReward \
      --check-rewards

Launch the 4xGPU GenRL HPSv3 + VideoAlign training run:

    WANDB_MODE=online \
    WANDB_ENTITY=<your-wandb-entity> \
    NUM_GPUS=4 \
    VIDEOALIGN_CHECKPOINT_PATH=.cache/VideoReward \
    FORCE_QWENVL_VIDEO_READER=opencv \
    PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True \
    bash examples/train/run.sh \
      examples/train/configs/rl/wan/genrl_hpsv3_videoalign.yaml \
      --training.checkpoint.output_dir outputs/genrl_longcat

For the 41-step reproduction probe:

    WANDB_MODE=online \
    WANDB_ENTITY=<your-wandb-entity> \
    NUM_GPUS=4 \
    VIDEOALIGN_CHECKPOINT_PATH=.cache/VideoReward \
    FORCE_QWENVL_VIDEO_READER=opencv \
    PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True \
    bash examples/train/run.sh \
      examples/train/configs/rl/wan/genrl_hpsv3_videoalign.yaml \
      --training.loop.max_train_steps 41 \
      --training.checkpoint.training_state_checkpointing_steps 20 \
      --training.checkpoint.output_dir outputs/genrl_longcat_repro_rewardfix_41

## Verification

Reward parity was checked against the previous successful GenRL run source state:

- FastVideo commit: `17aecbe2dd07245333a1c0ea85f89b2b7a4a1f88`
- HPSv3 submodule: `a2eb2ef2c7b5d91a566347a5825cf6d872122149`
- VideoAlign submodule: `aba26b658fec7d9fd30c295187b548ea673c8769`
- VideoReward checkpoint: `checkpoint-11352`

Fixed-video reward parity after syncing the vendored runtime:

    hpsv3_general delta:    ~1e-6
    hpsv3_percentile delta: ~1e-6
    videoalign_mq delta:    small residual runtime drift, about -0.03 to +0.04
    videoalign_ta delta:    small residual runtime drift, about -0.04 to +0.02

Training reproduction run after the reward-runtime fix showed reward curves recovering compared with the earlier bad run.

## Test Plan

    python -m py_compile \
      examples/train/prepare_genrl_assets.py \
      fastvideo/train/methods/rl/utils/data.py \
      fastvideo/train/methods/rl/reward/hpsv3.py \
      fastvideo/train/methods/rl/reward/videoalign.py

    python examples/train/prepare_genrl_assets.py --help

    python -c "import yaml; from pathlib import Path; p=Path('examples/train/configs/rl/wan/genrl_hpsv3_videoalign.yaml'); yaml.safe_load(p.read_text()); print(f'parsed {p}')"

## Test Results

    python -m py_compile ...  # passed
    python examples/train/prepare_genrl_assets.py --help  # passed
    YAML parse check  # passed

`pre-commit` was not available in the local shell or the `fastvideo` conda env, so I could not run the full hook set locally.

## Checklist

- [ ] I ran `pre-commit run --all-files` and fixed all issues
- [x] I added or updated tests / validation for my changes
- [x] I updated documentation if needed
- [x] I considered GPU memory impact of my changes

### CODE REVIEW

**Review by SolitaryThinker — CHANGES_REQUESTED** [review]
Requesting changes at exact head `03f3e8913cc6f347538ba1dcf8836f6a133b85c7`.

This is not a docs-only increment: it vendors and packages two reward runtimes on top of the still-open `py/add_rl` integration branch.

- **[P1] `examples/train/requirements-genrl.txt:6` / `examples/train/README.md:104`**: the documented sequential installs cannot produce a valid environment. This head requires `datasets>=4.0.0` and then pins `datasets==3.6.0`; against current main it also pins Transformers 4.x versus `>=5` and Diffusers 0.33 versus `>=0.38`. The README additionally points at a Torch 2.10 FlashAttention wheel while the requirements select Torch 2.11. Please provide one compatible, tested dependency contract.
- **[P1] `fastvideo/train/methods/rl/reward/HPSv3/hpsv3/inference.py:35`**: runtime loading opens `config/HPSv3_7B.yaml`, but a clean wheel built from this exact head contains no copy of that file. Installed reward loading therefore fails. Include the required package data and add an installed-wheel smoke test.
- **[P1] vendored `HPSv3` and `VideoAlign` runtimes**: both files identify MIT-licensed upstream code, but neither upstream license notice is included; the built wheel contains only the FastVideo license. Preserve the upstream notices with the copied source.
- **[P1] `examples/train/prepare_genrl_assets.py:30,135` and `HPSv3/hpsv3/inference.py:38`**: the advertised reproducible setup follows moving GitHub and Hugging Face heads, and VideoAlign silently selects the latest checkpoint. Pin repository and model revisions/checkpoints, verify them, and record the resolved identities.
- **[P2] `prepare_genrl_assets.py:282`**: custom asset destinations are merely printed. `PROMPT_DATASET_PATH` is never consumed, the YAML hardcodes `.cache/genrl_filtered_prompts`, and environment changes cannot survive helper exit. Pass the selected paths into the actual launch/config surface.
- **[P2] `prepare_genrl_assets.py:149` versus `VideoAlign/utils.py:174`**: validation accepts root-level model files, while the loader only accepts `checkpoint-*` directories. The helper can report ready for an unusable layout; validate the exact loader contract.

Static checks passed: Python compilation, helper `--help`, YAML parsing, launcher shell syntax, prompt fetch/validation, and clean wheel construction. GPU reward preflight and full training were not available locally.

Finally, this PR targets `py/add_rl@a520eceed7ad023b6107b209d8d4d7dc996fd370` (#1411), not `main`; that base is currently conflicting and substantially behind main. The stack needs a current-main disposition before this PR can be reassessed.
